content/posts/redis-modules-into-arm-land-part1.md

```diff
@@ -0,0 +1,276 @@
+---
+title: "Getting Redis Modules Into ARM Land - Part 1"
+date: 2019-11-13
+authors:
+  - author:
+      name: "Rafi Einstein"
+      link: "https://github.com/rafie"
+tags: ["redisedge", "arm"]
+---
+
+Here at Redis Labs, RedisEdge was our motivation for getting into ARM land and dragging Redis modules along.
+Redis, of course, has long been native in this land, in both glibc and alpine/musl variants.
+Redis modules have already been on the multi-platform scene, running on various Linux distributions and supporting macOS mainly for the sake of development experience. However, it was more enterprise/data-center-oriented until RedisEdge, which targets IoT devices.
+In this series of posts, I'll describe our vision of ARM platform support and the developer user experience, as well as the steps we took to get there.
+
+If you follow along, you'll end up with a fully functional ARM build laboratory.
+
+## Inside RedisEdge
+
+![redisedge1](/redis-modules-arm-1.png)
+
+But first, let's take a look at RedisEdge.
+RedisEdge is not a Redis module, but an aggregate of three Redis modules: RedisGears, RedisAI, and RedisTimeSeries.
+It is distributed as a Docker image, which is based on Redis Server 5.0. Thus, one can simply pull the image, run it, and start issuing Redis commands, load models into RedisAI, and execute Python gears scripts on RedisGears.

```

Comment:
If this is to distinguish the three different sets of actions of the three Redis modules, let's change to "Thus, one can simply pull the image, run it, and start issuing Redis commands; load models into RedisAI; and execute Python gears scripts on RedisGears."

Reply:
I was trying to give the sentence a sort of conversational nature. Does the use of semicolons interfere with that?

Reply:
It sounds like you're trying to describe three sets of actions: 
- pull the image, run it, and start issuing Redis commands
- load models into RedisAI
- execute Python gears scripts on RedisGears

So to distinguish the three groups of actions I put the semi-colons in, because otherwise there's a lot of commas and it can be hard to distinguish the separate actions. 

If we don't necessarily need to group the "pull the image, run it, and start issuing Redis commands" together, we can rephrase to "Thus, one can simply pull the image and run it, start issuing Redis commands, load models into RedisAI, and execute Python gears scripts on RedisGears."